rust/perspective-viewer/src/rust/components/column_settings_sidebar/style_tab.rs

```diff
@@ -0,0 +1,224 @@
+// ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+// ┃ ██████ ██████ ██████       █      █      █      █      █ █▄  ▀███ █       ┃
+// ┃ ▄▄▄▄▄█ █▄▄▄▄▄ ▄▄▄▄▄█  ▀▀▀▀▀█▀▀▀▀▀ █ ▀▀▀▀▀█ ████████▌▐███ ███▄  ▀█ █ ▀▀▀▀▀ ┃
+// ┃ █▀▀▀▀▀ █▀▀▀▀▀ █▀██▀▀ ▄▄▄▄▄ █ ▄▄▄▄▄█ ▄▄▄▄▄█ ████████▌▐███ █████▄   █ ▄▄▄▄▄ ┃
+// ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+// ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+// ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+// ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+// ┃ This file is part of the Perspective library, distributed under the terms ┃
+// ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+// ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::{JsCast, JsValue};
+use yew::{function_component, html, Callback, Html, Properties};
+
+use crate::components::datetime_column_style::DatetimeColumnStyle;
+use crate::components::number_column_style::NumberColumnStyle;
+use crate::components::string_column_style::StringColumnStyle;
+use crate::components::style::LocalStyle;
+use crate::config::Type;
+use crate::presentation::Presentation;
+use crate::renderer::Renderer;
+use crate::session::Session;
+use crate::utils::JsValueSerdeExt;
+use crate::{clone, css, html_template};
+
+#[derive(Clone, PartialEq, Properties)]
+pub struct StyleTabProps {
+    pub session: Session,
+    pub renderer: Renderer,
+    pub presentation: Presentation,
+    pub column_name: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct DefaultConfig {
+    string: serde_json::Value,
+    datetime: serde_json::Value,
+    date: serde_json::Value,
+    integer: serde_json::Value,
+    float: serde_json::Value,
+    bool: serde_json::Value,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Config {
+    columns: HashMap<String, serde_json::Value>,
+    default_config: Option<DefaultConfig>,
+}
+
+/// This function sends the config to the plugin using its `restore` method
+fn send_config<T: Serialize + Debug>(
+    renderer: Renderer,
+    presentation: Presentation,
```

Comment:
If you pass these by reference, they will not need to be cloned at the call site.

The main place we want to `clone()` + move into a function is when spawning the first frame on a new async stack, as the async frame cannot reference the spawning stack.